web/server/www/scripts/productlist/ListOfProducts.js

```diff
@@ -149,6 +149,62 @@ function (declare, domClass, domBase, dom, ItemFileWriteStore, topic, DataGrid,
 
   //--- Product grid ---//
 
+  function formatProductStatus(item) {
+    if (!item.accessible) {
+      return '<span class="customIcon product-noaccess"></span>';
+    } else if (item.databaseStatus !== DBStatus.OK) {
+      return '<span class="customIcon product-error"></span>';
+    }
+    return '';
+  }
+
+  function formatProductIcon(item) {
+    var name = util.atou(item.displayedName_b64);
+
+    return '<div class="product-avatar" '
+      + 'style="background-color: '
+      + util.strToColorBlend(item.endpoint, "white", 0.75).toHex() + '">'
+      + '<span class="product-avatar">'
+      + name[0].toUpperCase()
+      + '</span></div>';
+  }
+
+  function formatProductName(item) {
+    var name = util.atou(item.displayedName_b64);
+
+    if (item.databaseStatus !== DBStatus.OK || !item.accessible) {
+      return '<span class="product-error">' + name + '</span>';
+    } else {
+      return '<span class="link">' + name + '</span>';
+    }
+  }
+
+  function formatProductDescription(item) {
+    var description = item.description_b64
+                      ? util.atou(item.description_b64)
+                      : "";
+
+      var dbStatus = item.databaseStatus;
+      var dbStatusMsg = util.dbStatusFromCodeToString(dbStatus);
```

Comment:
This string is only used in `dbStatus !== DBStatus.OK`, so the lookup could be deferred to that case.